src/internals/objects/Objects.ts

```diff
@@ -238,6 +238,38 @@ export namespace Objects {
     arg3 = unset
   > = Functions.PartialApply<CreateFn, [pattern, arg0, arg1, arg2, arg3]>;
 
+  type MutableImpl<
+    obj,
+    keys,
+    union = {
+      [key in keyof obj]: obj[key];
+    } & {
+      -readonly [key in Extract<keyof obj, keys>]: obj[key];
+    }
```

Comment:
`Extract<keyof obj, keys>` removes homomorphism so in case there were any property modifiers, it will remove them

The perfect example is optional property:

```ts
type res3 = Apply<
    //   ^?
    Objects.Mutable,
    [{ a: 1; readonly b: true; c?: "cc" }, "a" | "c"]
>;

type tes3 = Expect<Equal<res3, { a: 1; readonly b: true; c?: "cc" }>>;
```

So instead I suggest to use key remapping:

```suggestion
    union = {
      [key in keyof obj]: obj[key];
    } & {
      -readonly [key in keyof obj as key extends keys ? key : never]: obj[key];
    }
```

docs:
- homomorphism – https://stackoverflow.com/questions/59790508/what-does-homomorphic-mapped-type-mean/59791889#59791889
- key remapping – https://www.typescriptlang.org/docs/handbook/2/mapped-types.html#key-remapping-via-as